react/OrderItems.tsx

```diff
@@ -172,65 +182,76 @@ const useAddItemsTask = (
       mutationInputItems: OrderFormItemInput[]
       mutationInputMarketingData?: Partial<MarketingData>
       orderFormItems: Item[]
-    }) => () => {
-      return mutateAddItem({
-        variables: {
-          items: mutationInputItems,
-          marketingData: mutationInputMarketingData,
-        },
-      })
-        .then(({ data }) => data!.addToCart)
-        .then(updatedOrderForm => {
-          // update the uniqueId of the items that were
-          // added locally with the value from the server
-          orderFormItems.forEach(orderFormItem => {
-            const updatedItem = updatedOrderForm.items.find(
-              updatedOrderFormItem =>
-                updatedOrderFormItem.id === orderFormItem.id
-            )!
-
-            const fakeUniqueId = orderFormItem.uniqueId
-
-            // update all mutations in the queue that referenced
-            // this item with it's fake `uniqueId`
-            updateLocalQueueItemIds({
-              fakeUniqueId,
-              uniqueId: updatedItem.uniqueId,
-            })
-            fakeUniqueIdMapRef.current[fakeUniqueId] = updatedItem.uniqueId
-          })
+    }) => ({
+      execute: async () => {
+        const { data } = await mutateAddItem({
+          variables: {
+            items: mutationInputItems,
+            marketingData: mutationInputMarketingData,
+          },
+        })
```

Comment:
Any chance that this call fails? If there is, is that OK here?

Reply:
absolutely, the purpose of this PR is to handle the fail of this and the `update` mutation. if the mutation fails, the `rollback` function of the task will be called.